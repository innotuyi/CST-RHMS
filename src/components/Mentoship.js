import React, { useState } from "react";
import svg from "../images/Lecturesvg.svg";
import {
  Row,
  Col,
  Container,
  Form,
  Button,
  Image,
  Jumbotron,
} from "react-bootstrap";
import Header from "./Header";

function Mentoship() {


  const [name, setName] = useState("");
  const [title, setTitle] = useState("");
   const [university, setUniversity] = useState("");
   const [date, setDate] = useState("");

  const mentoHandler = (e) => {
    e.preventDefault();
    console.warn(name, title, university, date);
  };

    return (
      <div>
        <div
          style={{
            background: "linear-gradient(to bottom, white, 84%, #1089AF)",
          }}
        >
          <Header />
          <Container>
            <Row className="d-flex justify-content-center arighn-item-center">
              <Col className="leftimage">
                <Image
                  style={{
                    width: "80%",
                    height: "80%",
                    objectFit: "contain",
                    paddingTop: "80px",
                  }}
                  src={svg}
                />
              </Col>
              <Col className="d-flex justify-content-center">
                <Jumbotron
                  style={{
                    background:
                      "linear-gradient(to bottom, white , 90%, #1089AF )",
                    borderRadius: "10px",
                    width: "100%",
                  }}
                >
                  <Form className="form" style={{}} onSubmit={mentoHandler}>
                    <h2 className="text-center mb-5 text-secondary">
                      Add A Suppervised Mentoship
                    </h2>
                    <Form.Group>
                      <Form.Label>Enter Student's Name</Form.Label>
                      <Form.Control
                        required
                        type="Text"
                        id="name"
                        value={name}
                        onChange={(e) => setName(e.target.value)}
                        placeholder="Enter Student's Name"
                      />
                    </Form.Group>
                    <Form.Group>
                      <Form.Label>Enter The Project Title</Form.Label>
                      <Form.Control
                        required
                        type="Text"
                        placeholder="Enter The Project Title"
                        id="title"
                        value={title}
                        onChange={(e) => setTitle(e.target.value)}
                      />
                    </Form.Group>
                    <Form.Group>
                      <Form.Label>Enter The Student's University</Form.Label>
                      <Form.Control
                        required
                        type="Text"
                        placeholder="Enter The Student's University"
                        id="university"
                        value={university}
                        onChange={(e) => setUniversity(e.target.value)}
                      />
                    </Form.Group>
                    <Form.Group controlId="dob">
                      <Form.Label>Select The Date</Form.Label>
                      <Form.Control
                        type="date"
                        name="dos"
                        id="date"
                        value={date}
                        onChange={(e) => setDate(e.target.value)}
                        placeholder="Choose Date"
                      />
                    </Form.Group>
                    <Col className="text-center">
                      <Button className="" type="submit">
                        Submit
                      </Button>
                    </Col>
                  </Form>
                </Jumbotron>
              </Col>
            </Row>
          </Container>
        </div>
      </div>
    );
}

export default Mentoship
