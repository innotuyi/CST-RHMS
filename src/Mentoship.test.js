import React from "react";
import Mentoship from "./components/Mentoship";

import { configure, shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

let wrapper;
beforeEach(() => {
  wrapper = shallow(<Mentoship />);
});

describe("Mentoship Tests", () => {
  it("matches Snapshot Collectly", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

   it("should have All Inputs", () => {
     expect(wrapper.find("#name")).toHaveLength(1);
     expect(wrapper.find("#university")).toHaveLength(1);
     expect(wrapper.find("#date")).toHaveLength(1);
     expect(wrapper.find("#title")).toHaveLength(1);
   });
  
   it("should have a Button  Component", () => {
     expect(wrapper.find("Button")).toHaveLength(1);
     expect(wrapper.find("Button").text()).toEqual("Submit");
   });
  
  it("Should capture Name correctly onChange", () => {
    wrapper.find("#name").simulate("change", {
      target: {
        value: "Kwibuka",
      },
    });
    expect(wrapper.find("#name").props().value).toEqual("Kwibuka");
  });

   it("Should capture University correctly onChange", () => {
     wrapper.find("#university").simulate("change", {
       target: {
         value: "CST",
       },
     });
     expect(wrapper.find("#university").props().value).toEqual("CST");
   });
  
  it("Should capture Title correctly onChange", () => {
    wrapper.find("#title").simulate("change", {
      target: {
        value: "Research",
      },
    });
    expect(wrapper.find("#title").props().value).toEqual("Research");
  });

  it("Should capture Date correctly onChange", () => {
    wrapper.find("#date").simulate("change", {
      target: {
        value: "2021-05-19",
      },
    });
    expect(wrapper.find("#date").props().value).toEqual("2021-05-19");
  });
});
