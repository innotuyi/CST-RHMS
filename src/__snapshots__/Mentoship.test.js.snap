// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mentoship Tests matches Snapshot Collectly 1`] = `
<div>
  <div
    style={
      Object {
        "background": "linear-gradient(to bottom, white, 84%, #1089AF)",
      }
    }
  >
    <Header />
    <Container
      fluid={false}
    >
      <Row
        className="d-flex justify-content-center arighn-item-center"
        noGutters={false}
      >
        <Col
          className="leftimage"
        >
          <Image
            fluid={false}
            rounded={false}
            roundedCircle={false}
            src="Lecturesvg.svg"
            style={
              Object {
                "height": "80%",
                "objectFit": "contain",
                "paddingTop": "80px",
                "width": "80%",
              }
            }
            thumbnail={false}
          />
        </Col>
        <Col
          className="d-flex justify-content-center"
        >
          <Jumbotron
            fluid={false}
            style={
              Object {
                "background": "linear-gradient(to bottom, white , 90%, #1089AF )",
                "borderRadius": "10px",
                "width": "100%",
              }
            }
          >
            <Form
              className="form"
              inline={false}
              onSubmit={[Function]}
              style={Object {}}
            >
              <h2
                className="text-center mb-5 text-secondary"
              >
                Add A Suppervised Mentoship
              </h2>
              <FormGroup>
                <FormLabel
                  column={false}
                  srOnly={false}
                >
                  Enter Student's Name
                </FormLabel>
                <FormControl
                  id="name"
                  onChange={[Function]}
                  placeholder="Enter Student's Name"
                  required={true}
                  type="Text"
                  value=""
                />
              </FormGroup>
              <FormGroup>
                <FormLabel
                  column={false}
                  srOnly={false}
                >
                  Enter The Project Title
                </FormLabel>
                <FormControl
                  id="title"
                  onChange={[Function]}
                  placeholder="Enter The Project Title"
                  required={true}
                  type="Text"
                  value=""
                />
              </FormGroup>
              <FormGroup>
                <FormLabel
                  column={false}
                  srOnly={false}
                >
                  Enter The Student's University
                </FormLabel>
                <FormControl
                  id="university"
                  onChange={[Function]}
                  placeholder="Enter The Student's University"
                  required={true}
                  type="Text"
                  value=""
                />
              </FormGroup>
              <FormGroup
                controlId="dob"
              >
                <FormLabel
                  column={false}
                  srOnly={false}
                >
                  Select The Date
                </FormLabel>
                <FormControl
                  id="date"
                  name="dos"
                  onChange={[Function]}
                  placeholder="Choose Date"
                  type="date"
                  value=""
                />
              </FormGroup>
              <Col
                className="text-center"
              >
                <Button
                  active={false}
                  className=""
                  disabled={false}
                  type="submit"
                  variant="primary"
                >
                  Submit
                </Button>
              </Col>
            </Form>
          </Jumbotron>
        </Col>
      </Row>
    </Container>
  </div>
</div>
`;
